vumi/transports/turn/turn.py

```diff
@@ -0,0 +1,205 @@
+# -*- test-case-name: vumi.transports.turn.tests.test_turn -*-
+# -*- encoding: utf-8 -*-
+
+from urllib import urlencode
+from datetime import datetime
+import string
+import warnings
+from StringIO import StringIO
+
+from twisted.web import http
+from twisted.web.resource import Resource
+from twisted.web.server import NOT_DONE_YET
+from twisted.python import log
+from twisted.internet.defer import inlineCallbacks
+from twisted.internet.protocol import Protocol
+from twisted.internet.error import ConnectionRefusedError
+
+from vumi.utils import http_request_full, normalize_msisdn, LogFilterSite
+from vumi.transports.base import Transport
+from vumi.transports.failures import TemporaryFailure, PermanentFailure
+from vumi.errors import VumiError
+
+
+def iso8601(turn_timestamp):
+    # TODO: this will be a unix timestamp
+    if turn_timestamp:
+        ts = datetime.strptime(turn_timestamp, '%Y.%m.%d %H:%M:%S')
+        return ts.isoformat()
+    else:
+        return ''
+
+
+def normalize_outbound_msisdn(msisdn):
```

Comment:
Maybe a better name would be `format_msisdn_for_whatsapp` or something like that, since it takes the MSISDN, and formats it in the specific format for whatsapp.